common/src/main/java/org/wildfly/httpclient/common/ByteOutputs.java

```diff
@@ -73,4 +77,38 @@ public void write(final byte[] b, final int off, final int len) throws IOExcepti
 
     }
 
+    private static final class UnflushableByteOutput implements ByteOutput {
```

Comment:
Motivation?

Reply:
It is the way this protocol was implemented. Output is always sent at once to prevent fragmentation on the wire.
I maintained it for backward compatibility. For future version of the protocol we could revisit it if its really necessary.